replica/table.cc

```diff
@@ -508,10 +510,8 @@ void table::enable_off_strategy_trigger() {
 
 std::vector<std::unique_ptr<compaction_group>> table::make_compaction_groups() {
     std::vector<std::unique_ptr<compaction_group>> ret;
-    auto number_of_cg = 1 << _x_log2_compaction_groups;
-    ret.reserve(number_of_cg);
-    for (auto i = 0; i < number_of_cg; i++) {
-        ret.emplace_back(std::make_unique<compaction_group>(*this));
+    for (auto&& range : dht::split_token_range_msb(_x_log2_compaction_groups)) {
+        ret.emplace_back(std::make_unique<compaction_group>(*this, std::move(range)));
```

Comment:
auto ranges = dht::split_token_range_msb(_x_log2_compaction_groups);
ret.reserve(ranges.size());
...

with 256 groups, we avoid log2(growth_factor) 256 = 8 reallocations. but this is not a hot path, so this is not like urgent.